src/components/confirmationPopup.js

```diff
@@ -0,0 +1,28 @@
+import Popup from "./Popup";
+
+class confirmationPopup extends Popup {
+  constructor(popupSelector,  ) {
+    super({ popupSelector });
+  }
+  close(){
+
+    super.close();
+
+  }
+  open() {
+    super.open();
+  }
+  setSubmitAction(submitHandler){
+    this.submitHandler = submitHandler;
+
+  }
+  setEventListeners() {
+    super.setEventListeners()
+    const saveButton = this._popupElement.querySelector(".modal__save-button");
```

Comment:
**Could be improved**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` , `this._likeButton` and so on.